app/waf/src/services/bigip.service.ts

```diff
@@ -117,6 +117,53 @@ export class BigIpManager {
     throw new Error(`License not found: from ${resObj}`);
   }
 
+  async getConfigsyncIp(mgmtIp: string): Promise<string> {
```

Comment:
I think this function can be simplified. 
How about make it be simply get-and-return. 
For the function itself, its functionality just returns the information at that time. 

The `waitAndCheck` should be the work of the upper call's. 
If you were referring to getInterfaces. Please ignore that one, it should be changed too in my opinion.

Reply:
It is unlucky that I was referring to getInterfaces().... I hope DO task api can guarantee every setting is done, and we will not need any of these checkings eventually.

Reply:
However, I have no confidence on DO. 
You can leave it here. I'm planning to changing getInterfaces later.

Reply:
Let us open a request to DO for that.